client/src/pages/TablePage.jsx

```diff
@@ -45,12 +46,26 @@ const TablePage = () => {
       .catch((err) => console.log(`${err}`));
   };
 
+  const filterEntries = () => {
+    if (type === 'Expenses') {
+      return expenseData.filter((entry) => entry.name.includes(searchQuery));
+    } else if (type === 'Income') {
+      return incomeData.filter((entry) => entry.name.includes(searchQuery));
```

Comment:
Can we ignore case and extra whitespace? So for example: '    bEST bUy       ' would still match with 'Best Buy'

`data.filter(({ name }) => name.toLowerCase().includes(searchQuery?.trim().toLowerCase())`